test/functional/keyrings/aws_kms/test_client_suppliers.py

```diff
@@ -114,3 +134,24 @@ def test_allow_deny_nested_supplier():
         test_deny("us-west-2")
 
     excinfo.match("Unable to provide client for region 'us-west-2'")
+
+
+def test_deny_region_block_on_response():
+    region = "us-west-2"
+    test = DenyRegionsClientSupplier(
+        denied_regions=[region], client_supplier=AlwaysOneRegionClientSupplier(region_name=region)
+    )
+
+    # Catch the blocked region on response.
+    with pytest.raises(UnknownRegionError):
+        test(None)
+
+
+def test_allow_region_block_on_response():
+    test = AllowRegionsClientSupplier(
+        allowed_regions=[None, "us-west-2"], client_supplier=AlwaysOneRegionClientSupplier(region_name="us-east-2")
+    )
+
+    # Catch the blocked region on response.
+    with pytest.raises(UnknownRegionError):
+        test(None)
```

Comment:
Should we also test the positive cases? Passing None but having it be a valid region once it's checked?

Reply:
Good call. Additional tests added.